iceoryx_hoofs/posix/ipc/include/iox/detail/message_queue.inl

```diff
@@ -0,0 +1,220 @@
+                      .failureReturnValue(ERROR_CODE)
+                      // don't use the suppressErrorMessagesForErrnos method since QNX used EINTR instead of ETIMEDOUT
+                      .ignoreErrnos(TIMEOUT_ERRNO)
+                      .evaluate();
+
+    if (mqCall.has_error())
+    {
+        return err(errnoToEnum(mqCall.error().errnum));
+    }
+
+    if (mqCall->errnum == TIMEOUT_ERRNO)
+    {
+        return err(errnoToEnum(ETIMEDOUT));
+    }
+
+    return ok();
+}
+
+template <typename Type, MessageQueue::Termination Terminator>
+expected<void, PosixIpcChannelError> MessageQueue::sendImpl(not_null<const Type*> msg, uint64_t msgSize) const noexcept
+{
+    uint64_t msgSizeToSend = msgSize;
+    if constexpr (Terminator == Termination::NULL_TERMINATOR)
+    {
+        msgSizeToSend += NULL_TERMINATOR_SIZE;
+    }
+
+    if (msgSizeToSend > static_cast<uint64_t>(m_attributes.mq_msgsize))
+    {
+        IOX_LOG(ERROR, "the message which should be sent to the message queue '" << m_name << "' is too long");
+        return err(PosixIpcChannelError::MESSAGE_TOO_LONG);
+    }
+
+    auto mqCall =
+        IOX_POSIX_CALL(mq_send)(m_mqDescriptor, msg, msgSizeToSend, 1U).failureReturnValue(ERROR_CODE).evaluate();
+
+    if (mqCall.has_error())
+    {
+        return err(errnoToEnum(mqCall.error().errnum));
+    }
+
+    if (mqCall->errnum == TIMEOUT_ERRNO)
+    {
+        return err(errnoToEnum(ETIMEDOUT));
+    }
+
+    return ok();
+}
+
+template <typename Type, MessageQueue::Termination Terminator>
+expected<uint64_t, PosixIpcChannelError>
+MessageQueue::timedReceiveImpl(not_null<Type*> msg, uint64_t maxMsgSize, const units::Duration& timeout) const noexcept
+{
+    timespec timeOut = timeout.timespec(units::TimeSpecReference::Epoch);
+    auto mqCall = IOX_POSIX_CALL(mq_timedreceive)(m_mqDescriptor, msg, maxMsgSize, nullptr, &timeOut)
+                      .failureReturnValue(ERROR_CODE)
+                      // don't use the suppressErrorMessagesForErrnos method since QNX used EINTR instead of ETIMEDOUT
+                      .ignoreErrnos(TIMEOUT_ERRNO)
+                      .evaluate();
+
+    if (mqCall.has_error())
+    {
+        return err(errnoToEnum(mqCall.error().errnum));
+    }
+
+    if (mqCall->errnum == TIMEOUT_ERRNO)
+    {
+        return err(errnoToEnum(ETIMEDOUT));
+    }
+
+    const auto receivedMsgLength = static_cast<uint64_t>(mqCall->value);
+    if constexpr (Terminator == Termination::NULL_TERMINATOR)
+    {
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+        if (msg[receivedMsgLength - NULL_TERMINATOR_SIZE] != 0)
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+            msg[0] = 0;
+            return err(PosixIpcChannelError::INTERNAL_LOGIC_ERROR);
+        }
+        return ok<uint64_t>(receivedMsgLength - NULL_TERMINATOR_SIZE);
+    }
+
+    return ok<uint64_t>(receivedMsgLength);
```

Comment:
Could you refactor this to something like
```
return receiveVerification<Termination::NULL_TERMINATION>(&msg[0], receivedMsgLength);
```

Just in case this will be refactored in the future and one forgets to change the implementation in `receiveImpl`.

Reply:
Done!